structopt-derive/src/attrs.rs

```diff
@@ -208,6 +214,8 @@ impl Name {
                     ScreamingSnake => s.to_shouty_snake_case(),
                     Snake => s.to_snake_case(),
                     Verbatim => s,
+                    Lower => s.to_lowercase().replace("_", ""),
+                    Upper => s.to_uppercase().replace("_", ""),
```

Comment:
```suggestion
                    Lower => s.to_snake_case().replace("_", ""),
                    Upper => s.to_shouty_snake_case().replace("_", ""),
```

We can't rely on snake case for the origin.

Reply:
What other case could there by? Camel case would be converted when lowering and `-` are not allowed in variable names.

Reply:
Thinking about it, i forgot this also applied so environment variables so '-' could indeed be present.

Reply:
That reminded me - you have only changed `rename_all` but left `rename_all_env` untouched. I don't think this kind inconsistency does us any good; could you please tweak it as well?

Reply:
Now wait, it's changed. Nevermind.